File: 14. Longest Common Prefix.py
##ss
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        
        ans = 0
        mins = float('inf')
        for x in range(len(strs)):
            mins = min(mins,len(strs[x]))
        if len(strs) == 0 :
            return ""
        
        elif len(strs) == 1:
            return strs[0]
        
        for x in range(mins):
            if ans < mins:
                for y in range(1,len(strs)):
                    if strs[y][ans] != strs[y-1][ans]:
                        break

                if strs[y][ans] == strs[y-1][ans]:
                    ans+=1
                
        if ans == 0:
            return ""
        
        else:
            #return ans
            return strs[0][0:ans]
        

File: cpp_solutions/14. Longest Common Prefix.cpp
#include <iostream>
#include <algorithm>
using std::cout;
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans;
        int i = 0;
        int sizes = 200;
        for(int x= 0; x< strs.size();x++){
            if (strs[x].length() < sizes){
                sizes = strs[x].length();
            }
            //cout << strs[x].length();
        }
        
        for(int x = 0; x< sizes; x++){
            char compare = strs[0].at(x);
            bool prob = false;
            //cout << compare;
            for(i=0;i<strs.size();i++){
                if (strs[i].at(x)!=compare){
                    prob = true;
                    break;
                }
                
            }
            
            if(prob == true){
                return ans;
            }
            else{
                ans+=compare;
            }
        }
        return ans;
    }
};
